pkg/etcdcli/etcdcli.go

```diff
@@ -462,3 +426,60 @@ func (g *etcdClientGetter) Defragment(ctx context.Context, member *etcdserverpb.
 	}
 	return resp, nil
 }
+
+func endpoints(nodeLister corev1listers.NodeLister,
+	configmapsLister corev1listers.ConfigMapLister,
+	networkLister configv1listers.NetworkLister,
+	nodeListerSynced cache.InformerSynced,
+	configmapsListerSynced cache.InformerSynced,
+	networkListerSynced cache.InformerSynced) ([]string, error) {
+
+	if !nodeListerSynced() {
+		return nil, fmt.Errorf("node lister not synced")
+	}
+	if !configmapsListerSynced() {
+		return nil, fmt.Errorf("configmaps lister not synced")
+	}
+	if !networkListerSynced() {
+		return nil, fmt.Errorf("network lister not synced")
+	}
+
+	var etcdEndpoints []string
+	// This configmap should always only contain voting members. Learners should not be present here.
+	configmap, err := configmapsLister.ConfigMaps(operatorclient.TargetNamespace).Get("etcd-endpoints")
```

Comment:
would it make sense to consider only `NotFound`? Otherwise we should return an error

Reply:
good point, I'd still revert to listing nodes on transient issues too. Is there an error check for that too?

Reply:
> I'd still revert to listing nodes on transient issues too

why not to fail and retry?

Reply:
on a transient error we might build incorrectly the client and cache it for later

Reply:
The endpoints are updated whenever you're attempting to use the client, so I doubt it will be cached for longer than necessary. The catch-22 is that this is precisely the method that is being called for that, so IMHO it's better to keep going with potentially differently composed set of endpoints than degrading a controller while getting a client. The chances of trying to hit a learner member with a promote call at this point are tiny.